dotcom-rendering/src/web/components/Liveness.importable.tsx

```diff
@@ -0,0 +1,186 @@
+import { css } from '@emotion/react';
+import { useEffect, useState } from 'react';
+import { useApi } from '../lib/useApi';
+
+type Props = {
+	pageId: string;
+	webTitle: string;
+	ajaxUrl: string;
+};
+
+// TODO: Break this out into its own component
+const Toast = ({
+	onClick,
+	noOfNewPosts,
+}: {
+	onClick: () => void;
+	noOfNewPosts: number;
+}) => {
+	// TODO: Style and absolute position this component
+	return (
+		<nav
+			css={css`
+				position: fixed;
+				top: 20px;
+				left: 50px;
+				display: flex;
+				width: 100%;
+				align-items: center;
+			`}
+		>
+			<button onClick={onClick}>{`${noOfNewPosts} blah`}</button>
+		</nav>
+	);
+};
+
+const isServer = typeof window === 'undefined';
+
+function hydrateBlocks() {
+	console.log('hydrateBlocks');
+	// TODO Call existing hydration solution targetted to these elements
+	// Maybe we can make doHydration work for all gu-islands things
+	// where the gu-hydrated data attribute isn't set?
+}
+
+/**
+ * insertNewBlocks takes html and inserts it at the top of the liveblog
+ *
+ * @param {string} html The block html to be inserted
+ * @returns void
+ */
+function insertNewBlocks(html: string) {
+	const latestBlock = document.querySelector('#maincontent :first-child');
+	if (!latestBlock) return;
+	latestBlock.insertAdjacentHTML('beforebegin', html);
+}
+
+/**
+ * revealNewBlocks - style any blocks that have been inserted but are hidden such that
+ * they are revealed
+ */
+function revealNewBlocks() {
+	console.log('revealNewBlocks');
+}
+
+/**
+ * This abstracts the code to construct the url we use for polling. Of note is the fact
+ * we have put `latestBlockId` in react state; by doing this we are able to change the url
+ * that SWR uses for the request whilst still benefitting from the polling logic
+ *
+ * @returns string
+ */
+function getKey(
+	pageId: string,
+	ajaxUrl: string,
+	filterKeyEvents: boolean,
+	latestBlockId: string,
+): string {
+	// Construct the url to poll
+	const url = new URL(`${pageId}.json`, ajaxUrl);
```

Comment:
My guess here is this is unlikely (maybe impossible?) to ever happen as both pageId and ajaxUrl are required as strings but I've added extra code to handle this just in case